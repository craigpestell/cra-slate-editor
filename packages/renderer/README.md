# @canner/slate-editor-renderer

[![npm version](https://badge.fury.io/js/%40canner%2Fslate-editor-renderer.svg)](https://badge.fury.io/js/%40canner%2Fslate-editor-renderer)

**NOTE: You should directly use it's plugin instead of writing render function.**

```js
import renderer from "@canner/slate-editor-renderer";
```

Renderers that define corresponding nodes and marks while rendering your editor.

methods:

### commonNode(TagName) => React.Element<TagName>

If you want to render general tag names for nodes such as `p`, `blockquote`, `h1`, `h2`, etc...

### commonMark(TagName, styleKey?, styleValue?)

If you want to render general tag names for nodes such as `span`, `strong`, `code`, `i`, etc...

### emojiNode()

rendering emoji

### imageNode()

rendering video

### linkNode()

renderling inline links

### videoNode(source: 'youtube' | 'dailymotion' | 'youku' | 'vimeo')

rendering video

Supported video sources:

* youtube
* dailymotion
* youku
* vimeo

### Complete example:

```js
// all cases
function renderMark(props) {
  // pass this function to your editor's renderMark prop
  switch (props.mark.type) {
    case "bold":
      return commonMark("strong")(props);
    case "code":
      return commonMark("code")(props);
    case "fontBgColor":
      return commonMark("span", "backgroundColor", "color")(props);
    case "fontColor":
      return commonMark("span", "color", "color")(props);
    case "fontSize":
      return commonMark("span", "fontSize")(props);
    case "letterSpacing":
      return commonMark("span", "letterSpacing")(props);
    case "italic":
      return commonMark("i")(props);
    case "strikethrough":
      return commonMark("s")(props);
    case "underline":
      return commonMark("u")(props);
  }
}

function renderNode(props) {
  // pass this function to your editor's renderNode prop
  switch (props.node.type) {
    case "paragraph":
      return commonNode("p")(props);
    case "blockquote":
      return commonNode("blockquote")(props);
    case "emoji":
      return emojiNode()(props);
    case "heading1":
      return commonNode("h1")(props);
    case "heading2":
      return commonNode("h2")(props);
    case "list-ul":
      return commonNode("ul")(props);
    case "list-ol":
      return commonNode("ol")(props);
    case "list-item":
      return commonNode("li")(props);
    case "image":
      return imageNode()(props);
    case "link":
      return linkNode()(props);
    case "youtube":
      return videoNode("youtube")(props);
    case "dailymotion":
      return videoNode("dailymotion")(props);
    case "youku":
      return videoNode("youku")(props);
    case "vimeo":
      return videoNode("vimeo")(props);
  }
}
```

# @canner/slate-editor-html

Serializer and deserializer
