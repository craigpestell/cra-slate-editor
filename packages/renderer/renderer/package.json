{
  "name": "@canner/slate-editor-renderer",
  "version": "1.5.0",
  "description": "Renderers for slate nodes and marks",
  "main": "lib/index.js",
  "author": "",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "peerDependencies": {
    "react": "^0.14.0 || ^15.0.0 || 16.x",
    "react-dom": "^0.14.0 || ^15.0.0 || 16.x",
    "slate": "^0.33.0",
    "styled-components": "^3.2.1"
  },
  "dependencies": {
    "@canner/slate-helper-inline-adddata": "^1.5.0",
    "emoji-mart": "^2.5.0",
    "js-video-url-parser": "^0.2.7",
    "lodash.mapvalues": "^4.6.0",
    "prismjs": "^1.15.0",
    "quill-icons": "^1.5.0",
    "react-icons": "^2.2.7",
    "react-loading-image": "^0.5.0",
    "react-resizable": "^1.7.5",
    "slate-edit-table": "^0.15.0-alpha.4"
  },
  "gitHead": "bb30f8746be7cbe2cda641b6b74c9d976bb56e31"
}
