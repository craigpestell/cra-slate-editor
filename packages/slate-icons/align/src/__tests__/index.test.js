import React from "react";
import { Change, Value } from "slate";
import { join } from "path";
import { AlignCenter, AlignLeft, AlignRight } from "../";
import iconTest from "test/icon-test";
import renderer from "react-test-renderer";
const onChange = arg => arg;

test("create a AlignCenter icon", () => {
  const component = renderer.create(
    <AlignCenter
      change={new Change({ value: new Value() })}
      onChange={onChange}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("create a AlignLeft icon", () => {
  const component = renderer.create(
    <AlignLeft
      change={new Change({ value: new Value() })}
      onChange={onChange}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("create a AlignRight icon", () => {
  const component = renderer.create(
    <AlignRight
      change={new Change({ value: new Value() })}
      onChange={onChange}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("click AlignCenter icon", done => {
  iconTest({
    icon: AlignCenter,
    expectedPath: join(__dirname, "expected-center.yaml"),
    callback: done
  });
});

test("click AlignCenter icon use customized type", done => {
  iconTest({
    icon: AlignCenter,
    type: "customizeAlign",
    expectedPath: join(__dirname, "custom-expected-center.yaml"),
    callback: done
  });
});

test("click AlignLeft icon", done => {
  iconTest({
    icon: AlignLeft,
    expectedPath: join(__dirname, "expected-left.yaml"),
    callback: done
  });
});

test("click AlignLeft icon use customized type", done => {
  iconTest({
    icon: AlignLeft,
    type: "customizeAlign",
    expectedPath: join(__dirname, "custom-expected-left.yaml"),
    callback: done
  });
});

test("click AlignRight icon", done => {
  iconTest({
    icon: AlignRight,
    expectedPath: join(__dirname, "expected-right.yaml"),
    callback: done
  });
});

test("click AlignRight icon use customized type", done => {
  iconTest({
    icon: AlignRight,
    type: "customizeAlign",
    expectedPath: join(__dirname, "custom-expected-right.yaml"),
    callback: done
  });
});
