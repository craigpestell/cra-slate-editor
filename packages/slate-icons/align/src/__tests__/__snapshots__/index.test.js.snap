// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a AlignCenter icon 1`] = `
<span
  data-active={false}
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <svg
    className={undefined}
    height="18px"
    onChange={[Function]}
    viewBox="0 0 18 18"
    width="18px"
  >
    <line
      className="ql-stroke"
      x1="15"
      x2="3"
      y1="9"
      y2="9"
    />
    <line
      className="ql-stroke"
      x1="14"
      x2="4"
      y1="14"
      y2="14"
    />
    <line
      className="ql-stroke"
      x1="12"
      x2="6"
      y1="4"
      y2="4"
    />
  </svg>
</span>
`;

exports[`create a AlignLeft icon 1`] = `
<span
  data-active={false}
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <svg
    className={undefined}
    height="18px"
    onChange={[Function]}
    viewBox="0 0 18 18"
    width="18px"
  >
    <line
      className="ql-stroke"
      x1="3"
      x2="15"
      y1="9"
      y2="9"
    />
    <line
      className="ql-stroke"
      x1="3"
      x2="13"
      y1="14"
      y2="14"
    />
    <line
      className="ql-stroke"
      x1="3"
      x2="9"
      y1="4"
      y2="4"
    />
  </svg>
</span>
`;

exports[`create a AlignRight icon 1`] = `
<span
  data-active={false}
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <svg
    className={undefined}
    height="18px"
    onChange={[Function]}
    viewBox="0 0 18 18"
    width="18px"
  >
    <line
      className="ql-stroke"
      x1="15"
      x2="3"
      y1="9"
      y2="9"
    />
    <line
      className="ql-stroke"
      x1="15"
      x2="5"
      y1="14"
      y2="14"
    />
    <line
      className="ql-stroke"
      x1="15"
      x2="9"
      y1="4"
      y2="4"
    />
  </svg>
</span>
`;
